### ISSUE TITLE
CA1062: False positive when passing parameter to a function

### ISSUE THREAD

**Issue by osolo** [dotnet/roslyn-analyzers#2668]
#### Analyzer package

Microsoft.CodeAnalysis.FxCopAnalyzers (https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.3 (Latest)

#### Diagnostic ID

Warning	CA1062	In externally visible method 'int Bar.DoSomething(Foo f)', validate parameter 'f' is non-null before using it. If appropriate, throw an ArgumentNullException when the argument is null or add a Code Contract precondition asserting non-null argument.	

#### Repro steps

When a checked parameter is passed to a function, there is a false positive:

```
    public class Foo
    {
        public int Val { get; set; }
    }

    public class Bar
    {
        public int DoSomething(Foo f)
        {
            if (f == null) { f = default; }
            f.Val++;  /* <-- OK, no warning */
            return DoSomethingElse(f); /* <--- false positive here */
        }
        private int DoSomethingElse(Foo f)
        {
            return f.Val;
        }
    }
```

#### Expected behavior
The parameter `f` above was checked for null, so there shouldn't be a warning.

Note that this is happening exclusively when passing the parameter to another function.  The line marked "OK" above shows no warnings (as expected).

This may be a duplicate of #2277, it's hard to tell.

**Comment by mavasani** [comment]
Yep, verified repros with https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers/2.9.3, but does not repro with https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers/2.9.4-beta1.final
